### PR TITLE
sam0/uart: merge samd21 & saml21 uart driver

### PR DESCRIPTION
**Pull request by dylad**
This PR merges UART driver for SAML21 & SAMD21 family.

This PR adds :
- updated periph_conf of all sam0 based boards
- common uart struct for all families
- Allow users to easily change GCLK source using periph_conf.h
- support for run in standby per sercom for both families
- add a second UART definition for SAML21 board

I tested this PR using examples/helloworld & tests/periph_uart on SAML21 Xpro.
I do not own a board based on SAMD21 so I'll require the help of someone to test it
and ensure nothing is broken

Don't hesitate to review and comment this PR ! 
Cheers !

### CODE REVIEW

File: cpu/sam0_common/periph/uart.c [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+/**
+ * @brief   Get the sercom gclk id of the given UART device
+ *
+ * @param[in] dev       UART device identifier
+ *
+ * @return              sercom gclk id
+ */
+
+static int _get_sercom_gclk_id(SercomUsart *dev)
```
Comment (dylad) [inline cpu/sam0_common/periph/uart.c]: Good point. I'll fix tonight.

File: cpu/sam0_common/periph/uart.c [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+    /* configure pins */
+    gpio_init(uart_config[uart].rx_pin, GPIO_IN);
+    gpio_init_mux(uart_config[uart].rx_pin, uart_config[uart].mux);
+    gpio_init(uart_config[uart].tx_pin, GPIO_OUT);
+    gpio_init_mux(uart_config[uart].tx_pin, uart_config[uart].mux);
+
+#ifdef CPU_FAM_SAMD21
+    /* calculate baudrate */
+    uint32_t  baud =  ((((uint32_t)CLOCK_CORECLOCK * 10) / baudrate) / 16);
```
Comment (aabadie) [inline cpu/sam0_common/periph/uart.c]: and after `=`

**Comment by dylad** [comment]
@kaspar030 @aabadie 
I addressed your comments.

File: boards/samr21-xpro/include/periph_conf.h [hunk 3]
```diff
@@ -117,15 +117,19 @@ static const uart_conf_t uart_config[] = {
         .tx_pin = GPIO_PIN(PA,4),
         .mux    = GPIO_MUX_D,
         .rx_pad = UART_PAD_RX_1,
-        .tx_pad = UART_PAD_TX_0
+        .tx_pad = UART_PAD_TX_0,
+        .runstdby = 0,
+        .gclk_src = GCLK_PCHCTRL_GEN_GCLK0
```
Comment (haukepetersen) [inline boards/samr21-xpro/include/periph_conf.h]: should be `GCLK_CLKCTRL_GEN_GCLK0`

File: boards/samr21-xpro/include/periph_conf.h [hunk 4]
```diff
@@ -117,15 +117,19 @@ static const uart_conf_t uart_config[] = {
...
     {
         .dev    = &SERCOM5->USART,
         .rx_pin = GPIO_PIN(PA,23),
         .tx_pin = GPIO_PIN(PA,22),
         .mux    = GPIO_MUX_D,
         .rx_pad = UART_PAD_RX_1,
-        .tx_pad = UART_PAD_TX_0
+        .tx_pad = UART_PAD_TX_0,
+        .runstdby = 0,
+        .gclk_src = GCLK_PCHCTRL_GEN_GCLK0
```
Comment (haukepetersen) [inline boards/samr21-xpro/include/periph_conf.h]: also `GCLK_CLKCTRL_GEN_GCLK0`

**Comment by dylad** [comment]
Nice catch @haukepetersen !

> I think there is quite a bit more potential unifying the uart drivers
I agree with you but there are some points we need to discuss with the community.

I fixed your comment & rebased :)

**Comment by haukepetersen** [comment]
needs rebase

**Comment by dylad** [comment]
Sorry about that, this is done.

**Comment by haukepetersen** [comment]
cppcheck is a little pissed about the `dev` variable that is only used in the `#ifdef`ed contexts. I suggest to fix it with substituting `s/dev/_uart(uart)/` throughout the code.

**Comment by dylad** [comment]
> Oh, just tried the samr21-xpro again: its not working anymore...

Whaaaaaat ? Hard times are coming...
I'll try to review but I have no hardware for this family.

**Comment by haukepetersen** [comment]
Will look into this later this evening and let you know

**Comment by haukepetersen** [comment]
false alarm, I seem to have done something wrong in my setup here -> everything runs fine.

ACK.

**Comment by haukepetersen** [comment]
and go

**Comment by dylad** [comment]
Many thanks @haukepetersen  !

**Pull request merged**